source/extensions/transport_sockets/tls/ssl_socket.h

```diff
@@ -94,6 +96,7 @@ class SslSocketInfo : public Envoy::Ssl::ConnectionInfo {
   mutable std::string cached_session_id_;
   mutable std::string cached_tls_version_;
   mutable SslExtendedSocketInfoImpl extended_socket_info_;
+  mutable absl::flat_hash_map<std::string, absl::optional<std::string>> cached_x509_extensions_;
```

Comment:
flat_hash_map doesn't have pointer stability for it's value, so this might cause the returned `const absl::optional<std::string>&` invalidate after the next `x509Extension` call.